src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,159 @@
+/* eslint-disable max-len */
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
+import React, { useEffect } from 'react';
+import { getPeople } from '../api';
+import { Person } from '../types';
+import { ErrorComponent } from './ErrorComponent';
 import { PeopleTable } from './PeopleTable';
+import { useSearchParams } from 'react-router-dom';
+
+export enum ErrorMessages {
+  BLANK = '',
+  NO_PEOPLE = 'There are no people on the server',
+  SOMETHING_WENT_WRONG = 'Something went wrong',
+  NO_PEOPLE_MATCHING_CRITERIA = 'There are no people matching the current search criteria',
+}
 
 export const PeoplePage = () => {
+  const [searchParams] = useSearchParams();
+  const [people, setPeople] = React.useState<Person[]>([]);
+  const [isLoading, setIsLoading] = React.useState(true);
+  const [error, setError] = React.useState<ErrorMessages>(ErrorMessages.BLANK);
+  const [filteredPeople, setFilteredPeople] = React.useState<Person[]>([]);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople()
+      .then(response => {
+        if (response.length === 0) {
+          setError(ErrorMessages.NO_PEOPLE);
+        }
+
+        setPeople(response);
+        setFilteredPeople(prev => [...prev, ...response]);
```

Comment:
The `setFilteredPeople` function is called with a function that spreads the previous state and adds the response. This might lead to duplicated entries if the effect runs multiple times. Consider setting `filteredPeople` directly to `response` instead of spreading `prev`.